### PR TITLE
Components

### PR DESCRIPTION
**Pull request by principles31**
# React capstone project - Metrics webapp
- The goal of this React capstone project is about building a mobile web application to check a list of metrics (numeric values) that you will create making use of React and Redux.
## Project requirements
API First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, you should implement it on your own.

> Some example APIs are:

- [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.
[Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.
- You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis) or in [ProgrammableWeb](https://www.programmableweb.com/category/all/apis). Some of the APIs requires a token for authentication and some others are just open.

> Design

- You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
- Colors (select one main color for your website).
- Typography: font face, size and weight.
- Layout: composition and space between elements.
- Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

> Interactions

> Home page

- When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- Along with the category name, you will display some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.
> Details page

- In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
> Testing requirements

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API, so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.
Technical requirements

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).
## In this project, the following were implemented
- Used the covid-19 API that allowed me to get numeric values based on a parameter.
- Followed these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)),
- When the page loads, the webapp shows the list of categories that can be filtered out by some parameter, for - example by the country name. And numeric values per country that come from the API. When the user clicks (or taps) on a country item, the application navigates to the details page.When the user clicks on the "Back" button (<), the user navigates to the home page.
- Created unit tests for pure functions
- A screenshot has been added 
> Desktop version
Screenshot from 2022-07-05 05-16-17
> Mobile version
Screenshot from 2022-07-05 05-17-33

-

### CODE REVIEW

File: src/components/NewsBody.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        </form>
+        <ul className="list-country">
+          {covidCountry.map((country) => (
+            <Link
+              className="a"
+              key={generate()}
+              to={{ pathname: `/country/${country.country}` }}
+            >
+              <li className="list-details">
+                <div className=" details">
```
Comment (RolandM99) [inline src/components/NewsBody.js:57]: - [x] I'll suggest you implement this feature below highlighted in the original design table picture. As part of this project, your project should look like the creative commons attribution design.

|Orginal Design|Your Design|
|----------------|--------------|
|image|image|

Kindly, implement those `circle-arrow` for each of your items or countries.

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @principles31, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/NewsBody.js (hunk 1 above)
Comment (principles31) [reply src/components/NewsBody.js:57]: @RolandM99 Thank you for the feedback
Please how do I implement those cirle arrows on lines 56-57 in Newsbody.js

**Comment by principles31** [comment]
- I'll suggest you implement this feature below highlighted in the original design table picture. As part of this project, your project should look like the creative commons attribution design.
- The above changes requested by the previous reviewer have been implemented to the best of my knowledge.

**Review by DammyShittu — APPROVED** [review]
Hi @principles31,

✔️ Homepage search functionality

## APPROVED 🟢 

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**